salt/_pillar/metalk8s_nodes.py

```diff
@@ -50,13 +53,100 @@ def node_info(node, ca_minion):
     return result
 
 
+def get_cluster_version(api_client):
+    coreV1 = kubernetes.client.CoreV1Api(api_client=api_client)
+    try:
+        namespace = coreV1.read_namespace(name="kube-system")
+    except (ApiException, HTTPError) as exc:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to read namespace information {}'.format(exc)
+        )
+    annotations = namespace.metadata.annotations
+    annotation_key = 'metalk8s.scality.com/cluster-version'
+    if not annotations or annotation_key not in annotations:
+        return __utils__['pillar_utils.errors_to_dict'](
+            'Unable to retrieve cluster version'
+        )
+
+    return namespace.metadata.annotations[annotation_key]
+
+
+def iso_timestamp_converter(timestamp):
+    if not timestamp:
+        return None
+    return timestamp.isoformat()
+
+
+def get_storage_classes(api_client):
+    storageV1 = kubernetes.client.StorageV1Api(api_client=api_client)
+    storage_classes = {}
+    try:
+        storageclass_list = storageV1.list_storage_class()
+    except (ApiException, HTTPError) as exc:
+        log.exception('Exception when calling storageV1->list_storage_class')
+        return None
+    for storageclass in storageclass_list.items:
+        # Need to convert the datetime object in storageclass to ISO format in
+        # order to make them serializable.
+        storageclass.metadata.creation_timestamp = iso_timestamp_converter(
+            storageclass.metadata.creation_timestamp
+        )
+        storageclass.metadata.deletion_timestamp = iso_timestamp_converter(
+            storageclass.metadata.deletion_timestamp
+        )
+        storage_classes[storageclass.metadata.name] = storageclass.to_dict()
+
+    return storage_classes
+
+
+def list_volumes(api_client, minion_id):
+    results = {}
+    customObjectApi = kubernetes.client.CustomObjectsApi(api_client=api_client)
+    storage_classes = get_storage_classes(api_client)
+
+    if not storage_classes:
```

Comment:
No, let's not do it this way. There are two **very** different cases covered by this test:

- `storage_classes is None`, which actually means that there was an API failure when we tried listing: this case should make `list_volumes` return a `pillar_utils.errors_to_dict`
- `storage_classes == {}`, which means there is no StorageClass in the API: this case can still be covered as it is right now (`return {}`)

Reply:
@gdemonet I think here, it was previously agreed upon that we will not log errors to pillar but rather use `log.exception` and return None

Reply:
It is fine